practice/3_python_testing/test_read_write_2_solved.py

```diff
@@ -0,0 +1,48 @@
+"""
+Write tests for python_part_2/task_read_write_2_solved.py task.
+To write files during tests use temporary files:
+https://docs.python.org/3/library/tempfile.html
+https://docs.pytest.org/en/6.2.x/tmpdir.html
+"""
+
+
+import tempfile
+import os
+import chardet
+from practice.python_part_2 import task_read_write_2_solved
+
+
+def is_valid_encoding(filepath, encoding):
+    try:
+        with open(filepath, 'rb') as f:
+            # Read a small chunk to determine encoding
+            sample = f.read(1024)
+        sample.decode(encoding, errors='strict')
+        return True
+    except (UnicodeDecodeError, IOError):
+        return False
+
+
+def test_rw2():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        task_read_write_2_solved.rw2(temp_dir)
+        # with open(os.path.join(temp_dir, 'file1.txt'), 'rb') as result_file1:
+        #     with open(os.path.join(temp_dir, 'file2.txt'), 'rb') as result_file2:
+        #         f1 = result_file1.read()
+        #         f2 = result_file2.read()
+        #
+        #         suggestion1 = chardet.detect(f1)
+        #         suggestion2 = chardet.detect(f2)
+        #
+        #         assert suggestion1['encoding'].lower() == 'utf-8'
+        #         assert suggestion2['encoding'].lower() == 'cp1252'
+        path_file1 = os.path.join(temp_dir, 'file1.txt')
+        path_file2 = os.path.join(temp_dir, 'file2.txt')
+        with open(path_file1, 'r') as result_file1:
+            with open(path_file2, 'r') as result_file2:
```

Comment:
Why do you have nested `with` and why assert is in `with?

Reply:
corrected